realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java

```diff
@@ -1923,4 +1925,103 @@ public void execute(Realm realm) {
             }
         });
     }
+
+    @Test
+    public void getRealmConfiguration_managedRealmObject() {
+        realm.beginTransaction();
+        AllTypes object = realm.createObject(AllTypes.class);
+        realm.commitTransaction();
+
+        assertSame(realmConfig, RealmObject.getConfiguration(object));
+    }
+
+    @Test
+    public void getRealmConfiguration_managedRealmModel() {
+        realm.beginTransaction();
+        AllTypesRealmModel object = realm.createObject(AllTypesRealmModel.class, 1L);
+        realm.commitTransaction();
+
+        assertSame(realmConfig, RealmObject.getConfiguration(object));
+    }
+
+    @Test
+    public void getRealmConfiguration_DynamicRealmObject() {
+        final DynamicRealm dynamicRealm = DynamicRealm.getInstance(realmConfig);
+        //noinspection TryFinallyCanBeTryWithResources
+        try {
+            dynamicRealm.beginTransaction();
+            DynamicRealmObject object = dynamicRealm.createObject("AllTypesRealmModel", 1L);
+            dynamicRealm.commitTransaction();
+
+            assertSame(realmConfig, RealmObject.getConfiguration(object));
+        } finally {
+            dynamicRealm.close();
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void getRealmConfiguration_unmanagedRealmObjectThrows() {
+        RealmObject.getConfiguration(new AllTypes());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void getRealmConfiguration_unmanagedRealmModelThrows() {
+        RealmObject.getConfiguration(new AllTypesRealmModel());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void getRealmConfiguration_null() {
+        RealmObject.getConfiguration(null);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void getRealmConfiguration_closedObjectThrows() {
+        realm.beginTransaction();
+        AllTypes object = realm.createObject(AllTypes.class);
+        realm.commitTransaction();
+
+        realm.close();
+        realm = null;
+
+        RealmObject.getConfiguration(object);
+    }
+
+    @Test(expected = IllegalStateException.class)
```

Comment:
We generate IllegalState and IllegalArgument exceptions for all manner of things.  I think all of these tests could be made more robust by checking to be sure that the Exception that is thrown is the one that you expect.